content/docs/features/dockerfiles.md

```diff
@@ -0,0 +1,315 @@
+    a sub-task of the `extender` and thus happens automatically after (and in the same container as) `extend`
+* Invoke the `exporter` as usual
+
+Extensions workflows are not currently supported when using the `creator` binary - support may be added in the future,
+but with a lower priority.
+
+### Risks
+
+Image extensions are considered experimental and susceptible to change in future API versions. Additionally, platform
+operators should be mindful that:
+
+* Dockerfiles are very powerful - in fact, you can do anything with a Dockerfile! Introducing image extensions into your
+  CNB builds can eliminate the security and compatibility guarantees that buildpacks provide if not done with great
+  care.
+* When Dockerfiles are used to switch the run image from that defined on the provided builder, the resulting run image
+  may not have all the mixins required by buildpacks that detected. Platforms may wish to optionally re-validate mixins
+  prior to `build` when using extensions.
+
+### Phased approach
+
+Some limitations of the initial implementation of the Dockerfiles feature have already been mentioned, and we'll expand
+on them here. As this is a large and complicated feature, the implementation has been split into phases in order to
+deliver incremental value and gather feedback.
+
+* Phase 1 (supported in lifecycle `0.15.0` or greater): one or more `run.Dockerfile`s each containing a single `FROM`
+  instruction can be used to switch the original run image to a new image (as no image modifications are permitted,
+  there is no need to run `extend` on the run image)
+* Phase 2 (supported in lifecycle `0.15.0` or greater): one or more `build.Dockerfile`s can be used to extend the
+  builder image
+  * A new `extend` lifecycle phase is introduced to apply `build.Dockerfile`s from `generate` to the builder image
+* Phase 3 (future): one or more `run.Dockerfile`s can be used to extend the run image
+  * The `extend` lifecycle phase can be run in parallel for the builder and run images
+
+The final ordering of lifecycle phases will look something like the following:
+
+* `analyze`
+* `detect` - after standard detection, `detect` will also run extensions' `./bin/generate`; output Dockerfiles are
+  written to a volume
+* `restore`
+* `extend` - applies one or more `build.Dockerfile`s to the builder image
+* `extend` - applies one or more `run.Dockerfile`s to the run image (could run in parallel with builder image extension)
+* `build`
+* `export`
+
+The [pack cli](https://github.com/buildpacks/pack)
+and [Tekton task](https://github.com/tektoncd/catalog/tree/main/task/buildpacks-phases/0.2) have or will eventually have
+support for builds using image extensions. However, [kpack](https://github.com/pivotal/kpack) - while able to use image
+extensions in builds - will need additional updating to propagate changes such as a switched run image back to the
+appropriate resource to avoid breaking rebase.
```

Comment:
Do you think readers of this document will be interested in technical details behinf `kpack` or will they be interested in a supported/not-supported decision?

```suggestion
extensions in builds - will need additional updating to propagate changes.
```

Reply:
I think a support/not-supported is what I'd care about and then a link to more details on how to enable it.

Reply:
I've restructured it a bit... LMK how it looks now